raft: TestNodeProposeWaitDropped failed

raft.TestNodeProposeWaitDropped [failed](https://tanzanite.cluster.engflow.com/invocations/default/3816f2eb-7981-4e9c-9412-4389178172ca?testReportRun=21&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvcmFmdDpyYWZ0X3Rlc3Q=) on release-24.2 @ [14962dfb7274e11754658d4fa2be32a1dd31b94b](https://github.com/cockroachdb/cockroach/commits/14962dfb7274e11754658d4fa2be32a1dd31b94b):


```
=== RUN   TestNodeProposeWaitDropped
raft2024/09/01 04:12:55 INFO: 1 switched to configuration voters=(1)
raft2024/09/01 04:12:55 INFO: 1 became follower at term 0
raft2024/09/01 04:12:55 INFO: newRaft 1 [peers: [1], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2024/09/01 04:12:55 INFO: 1 is starting a new election at term 0
raft2024/09/01 04:12:55 INFO: 1 became candidate at term 1
raft2024/09/01 04:12:55 INFO: 1 received MsgVoteResp from 1 at term 1
raft2024/09/01 04:12:55 INFO: 1 has received 1 MsgVoteResp votes and 0 vote rejections
raft2024/09/01 04:12:55 INFO: 1 became leader at term 1
raft2024/09/01 04:12:55 INFO: raft.node: 1 elected leader 1 at term 1
    node_test.go:379: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/raft/node_test.go:379
        	Error:      	Not equal: 
        	            	expected: *errors.errorString(&errors.errorString{s:"raft proposal dropped"})
        	            	actual  : context.deadlineExceededError(context.deadlineExceededError{})
        	Test:       	TestNodeProposeWaitDropped
    node_test.go:348: dropping message: type:MsgProp to:0 from:1 term:0 logTerm:0 index:0 entries:<Term:0 Index:0 Type:EntryNormal Data:"test_dropping" > commit:0 vote:0 reject:false rejectHint:0 match:0 
--- FAIL: TestNodeProposeWaitDropped (0.13s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=21</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestNodeProposeWaitDropped.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41809

--- a/pkg/raft/node_test.go
+++ b/pkg/raft/node_test.go
@@ -375,11 +375,7 @@ func TestNodeProposeWaitDropped(t *testing.T) {
 		}
 		n.Advance()
 	}
-	proposalTimeout := time.Millisecond * 100
-	ctx, cancel := context.WithTimeout(context.Background(), proposalTimeout)
-	// propose with cancel should be cancelled earyly if dropped
-	assert.Equal(t, ErrProposalDropped, n.Propose(ctx, droppingMsg))
-	cancel()
+	assert.Equal(t, ErrProposalDropped, n.Propose(context.Background(), droppingMsg))
 
 	n.Stop()
 	require.Empty(t, msgs)

